### PR TITLE
Format contact phone numbers more successfully

### PR DESCRIPTION
**Pull request by buoyad**
- Fetch country code from phone carrier info
- Use as default for contacts with no country code data
- Validate phone numbers in a way more similar to the server

cc @keybase/y2ksquad

### CODE REVIEW

File: shared/ios/Keybase/Utils.m [hunk 1]
```diff
@@ -7,35 +7,23 @@
 //
 
 #import "Utils.h"
+#import <CoreTelephony/CTCarrier.h>
+#import <CoreTelephony/CTTelephonyNetworkInfo.h>
 
 @implementation Utils
 
-// Returns YES if we are currently in a unit test context
-+ (BOOL)areWeBeingUnitTested {
```
Comment (buoyad) [inline shared/ios/Keybase/Utils.m]: cc @chrisnojima these seemed unused, ok to remove?

File: shared/actions/platform-specific/index.native.tsx [hunk 2]
```diff
@@ -507,14 +507,15 @@ async function manageContactsCache(
 
   // feature enabled and permission granted
   const contacts = await Contacts.getContactsAsync()
+  const defaultCountryCode = await NativeModules.Utils.getDefaultCountryCode()
   const mapped = contacts.data.reduce((ret: Array<RPCTypes.Contact>, contact) => {
     const {name, phoneNumbers = [], emails = []} = contact
 
     const components = phoneNumbers.reduce<RPCTypes.ContactComponent[]>((res, pn) => {
       // TODO this fails on many phone numbers, contact data from native may
```
Comment (jakob223) [inline shared/actions/platform-specific/index.native.tsx]: maybe update this TODO?

File: shared/actions/platform-specific/index.native.tsx [hunk 3]
```diff
@@ -546,6 +547,10 @@ async function manageContactsCache(
 const getE164 = (countryCode: string, phoneNumber: string) => {
   try {
     const parsed = phoneUtil.parse(phoneNumber, countryCode)
+    const reason = phoneUtil.isPossibleNumberWithReason(parsed)
```
Comment (jakob223) [inline shared/actions/platform-specific/index.native.tsx]: amusing misread: "is possible number within reason?"

File: shared/ios/Keybase/Utils.m (hunk 1 above)
Comment (buoyad) [reply shared/ios/Keybase/Utils.m]: Talked elsewhere; yes

File: shared/actions/platform-specific/index.native.tsx [hunk 4]
```diff
@@ -507,14 +507,17 @@ async function manageContactsCache(
 
   // feature enabled and permission granted
   const contacts = await Contacts.getContactsAsync()
+  let defaultCountryCode: string
+  try {
```
Comment (buoyad) [inline shared/actions/platform-specific/index.native.tsx]: cc @jakob223 @jzila I did some shuffling here to handle the potential `reject` in the android module

**Pull request merged**